### ISSUE TITLE
CoreOS gets stuck in loop if instance is created without a public interface

### ISSUE THREAD

**Issue by crawford** [coreos/bugs#596]
<a href="https://github.com/adamkdean"><img src="https://avatars.githubusercontent.com/u/1639527?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [adamkdean](https://github.com/adamkdean)**
_Thursday Dec 04, 2014 at 14:43 GMT_
_Originally opened as https://github.com/coreos/coreos-overlay/issues/981_

----

On Rackspace's public cloud, if I create a CoreOS instance without PublicNet, CoreOS gets stuck in a loop waiting for both eth0 and eth1 (https://github.com/coreos/coreos-overlay/blob/master/coreos-base/oem-rackspace/files/coreos-setup-environment).

Not every CoreOS instance wants to be sat on the internet, and I'd rather be able to sit behind a firewall that I route my traffic through. This is not easy with problems like this.

**Comment by crawford** [comment]
<a href="https://github.com/adamkdean"><img src="https://avatars.githubusercontent.com/u/1639527?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [adamkdean](https://github.com/adamkdean)**
_Thursday Dec 04, 2014 at 16:38 GMT_

----

One way around this is to create a dummy network, say 10.0.0.0 and put that in as eth1, and then change the etcd `addr` and `peer-addr` to `$public_ipv4`.

**Comment by crawford** [comment]
<a href="https://github.com/pwaller"><img src="https://avatars.githubusercontent.com/u/438648?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pwaller](https://github.com/pwaller)**
_Tuesday Apr 21, 2015 at 10:42 GMT_

----

Huh, I think I hit this too months ago.

**Comment by crawford** [comment]
We should be able to resolve this once we move to Ignition on Rackspace.

**Comment by ChrisMcKenzie** [comment]
Any word on this? Would really like to be able to spin coreos machines up without a hitch and this appears to be the last hitch.

**Comment by crawford** [comment]
@ChrisMcKenzie Unfortunately, no. This has been a fundamental issue with CoreOS on Rackspace for over two years now. Once we add support for Rackspace to Ignition, this should no longer be an issue, but that is not a very high priority for us right now.

**Comment by ChrisMcKenzie** [comment]
yeah currently what has to be done is boot the machine up login change the bash script to not look for eth1 and then restart the `coreos-setup-environment` service and everything is happy. would a pull request be welcome for this script if I figure out a clean way to detect the interfaces?

**Comment by crawford** [comment]
@ChrisMcKenzie Absolutely! We'll take all the help we can get with CoreOS on Rackspace.
